ociclient/client.go

```diff
@@ -552,3 +565,215 @@ func AddKnownMediaTypesToCtx(ctx context.Context, mediaTypes []string) context.C
+			if err != nil {
+				return ocispecv1.Descriptor{}, fmt.Errorf("unable to decompress layer blob: %w", err)
+			}
+
+			decompressedData, err := ioutil.ReadAll(decompressedReader)
+			if err != nil {
+				return ocispecv1.Descriptor{}, fmt.Errorf("unable to read decompressed layer blob: %w", err)
+			}
+
+			var mediatype string
+			switch decompressedReader.GetCompression() {
+			case compression.Uncompressed:
+				mediatype = ocispecv1.MediaTypeImageLayer
+			case compression.Gzip:
+				mediatype = ocispecv1.MediaTypeImageLayerGzip
+			case compression.Zstd:
+				mediatype = MediaTypeImageLayerZstd
+			}
+
+			des := ocispecv1.Descriptor{
+				Digest:    fslayer.BlobSum,
+				MediaType: mediatype,
+				Size:      int64(len(data)),
+			}
+
+			layers = append(layers, des)
+			decompressedDigests = append(decompressedDigests, digest.FromBytes(decompressedData))
+		}
+	}
+
+	configDesc, configBytes, err := createConfig(&v1Manifest, decompressedDigests, history)
+	if err != nil {
+		return ocispecv1.Descriptor{}, fmt.Errorf("unable to create config: %w", err)
+	}
+
+	v2ManifestDesc, v2ManifestBytes, err := createV2Manifest(configDesc, layers)
+	if err != nil {
+		return ocispecv1.Descriptor{}, fmt.Errorf("unable to create v2 manifest: %w", err)
+	}
+
+	err = c.cache.Add(configDesc, io.NopCloser(bytes.NewReader(configBytes)))
+	if err != nil {
+		return ocispecv1.Descriptor{}, fmt.Errorf("unable to write config blob to cache: %w", err)
+	}
+
+	err = c.cache.Add(v2ManifestDesc, io.NopCloser(bytes.NewReader(v2ManifestBytes)))
+	if err != nil {
+		return ocispecv1.Descriptor{}, fmt.Errorf("unable to write manifest blob to cache: %w", err)
+	}
+
+	return v2ManifestDesc, nil
+}
+
+func createV2Manifest(configDesc ocispecv1.Descriptor, layers []ocispecv1.Descriptor) (ocispecv1.Descriptor, []byte, error) {
+	v2Manifest := ocispecv1.Manifest{
+		Versioned: specs.Versioned{
+			SchemaVersion: 2,
+		},
+		Config: configDesc,
+		Layers: layers,
+	}
+
+	marshaledV2Manifest, err := json.MarshalIndent(v2Manifest, "", "   ")
+	if err != nil {
+		return ocispecv1.Descriptor{}, nil, fmt.Errorf("unable to marshal manifest: %w", err)
+	}
+
+	v2ManifestDesc := ocispecv1.Descriptor{
+		MediaType: ocispecv1.MediaTypeImageManifest,
+		Digest:    digest.FromBytes(marshaledV2Manifest),
+		Size:      int64(len(marshaledV2Manifest)),
+	}
+
+	return v2ManifestDesc, marshaledV2Manifest, nil
+}
+
+func createConfig(v1Manifest *v1Manifest, diffIDs []digest.Digest, history []ocispecv1.History) (ocispecv1.Descriptor, []byte, error) {
+	var config map[string]*json.RawMessage
+	if err := json.Unmarshal([]byte(v1Manifest.History[0].V1Compatibility), &config); err != nil {
+		return ocispecv1.Descriptor{}, nil, fmt.Errorf("unable to unmarshal config from v1 history: %w", err)
+	}
+
+	delete(config, "id")
+	delete(config, "parent")
+	delete(config, "Size") // Size is calculated from data on disk and is inconsistent
```

Comment:
why is it onconsistent

Reply:
I guess the comment is obsolete here. We just delete the `size` attribute because it isn't needed. Also the config size of v2 config is different due to a different data structure than the v1 config. I guess that is meant with inconsistency :>)